File: app/src/main/java/io/github/tonnyl/sample/MainActivity.java
package io.github.tonnyl.sample;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.text.Spannable;
import android.text.SpannableStringBuilder;
import android.text.style.StyleSpan;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;

import io.github.tonnyl.light.Light;

import static android.graphics.Typeface.BOLD_ITALIC;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        final FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);

        findViewById(R.id.button_success).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Light.success(fab, "Success", Light.LENGTH_SHORT)
                        .setAction("Action", new View.OnClickListener() {
                            @Override
                            public void onClick(View v) {
                                Toast.makeText(MainActivity.this, "Hello, Light!", Toast.LENGTH_SHORT).show();
                            }
                        })
                        .show();
            }
        });

        findViewById(R.id.button_info).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Light.info(fab, "Info", Light.LENGTH_SHORT).show();
            }
        });

        findViewById(R.id.button_warning).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Light.warning(fab, "Warning", Light.LENGTH_SHORT).show();
            }
        });

        findViewById(R.id.button_error).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Light.error(fab, "Error", Light.LENGTH_SHORT).show();
            }
        });

        findViewById(R.id.button_normal).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String prefix = "Formatted ";
                String highlight = "bold italic";
                String suffix = " text";
                SpannableStringBuilder ssb = new SpannableStringBuilder(prefix).append(highlight).append(suffix);
                int prefixLen = prefix.length();
                ssb.setSpan(new StyleSpan(BOLD_ITALIC),
                        prefixLen, prefixLen + highlight.length(), Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
                Light.make(fab, ssb, Light.TYPE_NORMAL, Light.LENGTH_SHORT).show();
            }
        });

        findViewById(R.id.button_custom).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Light.make(
                        fab,
                        "Awesome Snackbar",
                        R.drawable.ic_album_white_24dp,
                        R.color.color_cyan,
                        android.R.color.white,
                        Light.LENGTH_INDEFINITE,
                        R.drawable.ic_done_all_white_24dp,
                        R.color.colorAccent)
                        .setAction("Done all", new View.OnClickListener() {
                            @Override
                            public void onClick(View v) {
                                Toast.makeText(MainActivity.this, "Hello, Light!", Toast.LENGTH_SHORT).show();
                            }
                        })
                        .show();
            }
        });

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: sample/src/main/java/io/github/tonnyl/sample/MainActivity.kt
/*
 * The MIT License (MIT)
 *
 * Copyright (c) 2017 Lizhaotailang
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package io.github.tonnyl.sample

import android.content.Intent
import android.graphics.Typeface.BOLD_ITALIC
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity
import android.text.Spannable
import android.text.SpannableStringBuilder
import android.text.style.StyleSpan
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import io.github.tonnyl.light.*
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.content_main.*

class MainActivity : AppCompatActivity() {

    private var position = BOTTOM

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        button_success.setOnClickListener {
            success(fab, R.string.success, Snackbar.LENGTH_SHORT, position)
                    .setAction("Action", {
                        Toast.makeText(this@MainActivity, "Hello, Light!", Toast.LENGTH_SHORT).show()
                    })
                    .show()
        }

        button_info.setOnClickListener {
            info(fab, R.string.info, Snackbar.LENGTH_SHORT, position).show()
        }

        button_warning.setOnClickListener {
            warning(fab, R.string.warning, Snackbar.LENGTH_SHORT, position).show()
        }

        button_error.setOnClickListener {
            error(fab, R.string.error, Snackbar.LENGTH_SHORT, position).show()
        }

        button_normal.setOnClickListener {
            val prefix = "Formatted "
            val highlight = "bold italic"
            val suffix = " text"
            val ssb = SpannableStringBuilder(prefix)
                    .append(highlight)
                    .append(suffix)
                    .apply {
                        setSpan(StyleSpan(BOLD_ITALIC), prefix.length, prefix.length + highlight.length, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
                    }
            normal(fab, ssb.toString(), Snackbar.LENGTH_SHORT, position).show()
        }

        button_custom.setOnClickListener {
            make(
                    fab,
                    "Awesome Snackbar",
                    Snackbar.LENGTH_SHORT,
                    R.drawable.ic_album_white_24dp,
                    R.color.color_cyan,
                    android.R.color.white,
                    R.drawable.ic_done_all_white_24dp,
                    R.color.colorAccent, position)
                    .setAction("Done all", {
                        Toast.makeText(this@MainActivity, "Hello, Light!", Toast.LENGTH_SHORT).show()
                    })
                    .show()
        }

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_settings -> {
                startActivity(Intent(this@MainActivity, SettingsActivity::class.java))
            }
            else -> {
                position = if (position == BOTTOM) {
                    item.title = "TOP"
                    TOP
                } else {
                    item.title = "BOTTOM"
                    BOTTOM
                }
            }
        }
        return true
    }

}
